apps/callscreen/js/calls_handler.js

```diff
@@ -209,7 +204,8 @@ var CallsHandler = (function callsHandler() {
   function removeCall(index) {
     handledCalls.splice(index, 1);
 
-    if (handledCalls.length === 0) {
+    if ((handledCalls.length === 0) ||
+        (handledCalls[0].call.state === 'disconnected')) {
```

Comment:
If we have two calls hung up in rapid succession, this will be insufficient. If we want to be pedantic, we should write something like this instead:
```js
if (handledCalls.length === 0 ||
    !!handledCalls.find(call => call.state !== 'disconnected')) {
  // ...
```